src/processes/helper/userIdsToProcess.js

```diff
@@ -23,87 +42,88 @@ const findF2UserIds = async (userId) => {
       ]
     },
     limit: 1000,
     attributes: ['user_id_followed'],
-    raw : true
-  }).then( users => users.map(user => user.user_id_followed));
+    raw: true
+  }).then((users) => users.map((user) => user.user_id_followed));
 
-  return f2UserIds
-}
-
-const findBlockedUserIds = async (userId) => {
-  const blockedUserIds = await UserBlockedUser.findAll({
-    where: {
-      user_id_blocker: userId,
-    },
-    limit: 1000,
-    attributes: ['user_id_blocked'],
-    raw : true
-  }).then( users => users.map(user => user.user_id_blocked));
-
-  return blockedUserIds
-}
+  return f2UserIds;
+};
 
 const findRelatedUserIds = async (userId) => {
   const [followedByUserId, f2UserIds, blockedUserIds] = await Promise.all([
     findFollowingUserIds(userId),
     findF2UserIds(userId),
-    findBlockedUserIds(userId),
+    findBlockedUserIds(userId)
   ]);
 
-  let listUser = [...followedByUserId, ...f2UserIds, ...blockedUserIds];
-  let listUniqueUser = [...new Set(listUser)]
-  return listUniqueUser
-}
+  const listUser = [...followedByUserId, ...f2UserIds, ...blockedUserIds];
+  const listUniqueUser = [...new Set(listUser)];
+  return listUniqueUser;
+};
 
-const findUnrelatedUserIds = async (userId) => {
-  const relatedUserIds = await findRelatedUserIds(userId)
+const findUnrelatedUserIds = async (relatedUserIds) => {
   const unrelatedUserIds = await User.findAll({
     where: {
-      user_id: { [Op.notIn]: relatedUserIds }
+      user_id: {[Op.notIn]: relatedUserIds},
+      is_anonymous: false
     },
     limit: 1000,
     attributes: ['user_id'],
-    raw : true
-  }).then( users => users.map(user => user.user_id));
+    raw: true
+  }).then((users) => users.map((user) => user.user_id));
 
-  return unrelatedUserIds
-}
+  return unrelatedUserIds;
+};
+
+const findUnrelatedAnonUserIds = async (relatedUserIds) => {
+  const unrelatedUserIds = await User.findAll({
+    where: {
+      user_id: {[Op.notIn]: relatedUserIds},
+      is_anonymous: true
+    },
+    limit: 1000,
+    attributes: ['user_id'],
+    raw: true
+  }).then((users) => users.map((user) => user.user_id));
+
+  return unrelatedUserIds;
+};
 
 const findFollowingTopicByUser = async (userId) => {
   // TODO
   // Simplify this logic using model associate
   const followingTopicByUser = await UserTopic.findAll({
     where: {
-      user_id: userId,
+      user_id: userId
     },
     limit: 1000,
     attributes: ['topic_id'],
-    raw : true
-  }).then( topics => topics.map(topic => topic.topic_id ));
+    raw: true
+  }).then((topics) => topics.map((topic) => topic.topic_id));
 
   const topicList = await Topics.findAll({
     where: {
-      topic_id: { [Op.in]: followingTopicByUser },
+      topic_id: {[Op.in]: followingTopicByUser}
     },
     limit: 1000,
     attributes: ['name'],
-    raw : true
-  }).then( topics => topics.map(topic => topic.name ));
+    raw: true
+  }).then((topics) => topics.map((topic) => topic.name));
 
   return topicList;
-}
-
+};
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

In the `findFollowingTopicByUser` function, you're making two separate queries to get the topic names followed by a user. You could optimize this by joining the `UserTopic` and `Topics` tables in a single query to directly get the topic names.



<!-- This is an auto-generated comment by OSS CodeRabbit -->